docs/topics/using/timeouts.md

```diff
@@ -4,15 +4,25 @@ Ambassador Edge Stack enables you to control timeouts in several different ways.
 
 ## Request Timeout: `timeout_ms`
 
-`timeout_ms` is the timeout for an entire user-level transaction. By default, this is 3000ms. This spans the point at which the entire downstream request has been processed (i.e., end of stream) to the point where the upstream response has been processed. This timeout includes all retries.
+`timeout_ms` is the end-to-end timeout for an entire user-level transaction. It begins after the full incoming request is received up until the full response stream is returned to the client. This timeout includes all retries. By default, this is 3000ms.  It can be disabled by setting the value to 0.
 
 ## Idle Timeout: `idle_timeout_ms`
 
-`idle_timeout_ms` controls how long a connection should remain open when no traffic is being sent through the connection. If not set, Ambassador Edge Stack will default to the value set by `cluster_idle_timeout_ms`
+`idle_timeout_ms` controls how long a connection should remain open when no traffic is being sent through the connection. `idle_timeout_ms` is distinct from `timeout_ms`, as the idle timeout applies on either down or upstream request events and is reset every time an encode/decode event occurrs or data is processed for the stream. `idle_timeout_ms` operates on a per-route basis and will overwrite behavior of the `cluster_idle_timeout_ms`.  If not set, Ambassador Edge Stack will default to the value set by `cluster_idle_timeout_ms`. It can be disabled by setting the value to 0.
+
+## Cluster Idle Timeout: `cluster_idle_timeout_ms`
+
+`cluster_idle_timeout_ms` controls how long a connection stream will remain open if there are no active requests. This timeout operates based on outgoing requests to upstream services. It can be disabled by setting the value to 0.
 
 ## Connect Timeout: `connect_timeout_ms`
 
-`connect_timeout_ms` controls the connection-level timeout for Ambassador Edge Stack to an upstream service. The default is `3000m`.
+`connect_timeout_ms` sets the connection-level timeout for Ambassador Edge Stack to an upstream service at the network layer.  This timeout runs until Ambassador can verify that a TCP connection has been established.  This timeout cannot be disabled. The default is `3000m`.
```

Comment:
...3000 _meters_? Should probably be just "3000". (I know you didn't change that but we should fix it. 😉 )

Reply:
haha, good point.  Fixed, also added some of the defaults for others (as milliseconds).